Add rundeck cli and project management

Joris29 opened the pull request:
#### Pull Request (PR) description
This introduces the rundeck cli as a seperate class and als add a define to create and manage projects

Fixes #449
Fixes #181
Fixes #67

Joris29 commented:
@kenyon Hello it's been a while but i wanted to propose this first draft of the changes i added.
If you have questions or proposals feel free to add them.

kenyon commented on data/Debian.yaml:
```diff
@@ -9,3 +9,5 @@ rundeck::repo_config:
     key:
       name: rundeck
```
This should fix the repo signature validation problem (should be a separate PR, but I noticed it here):

```suggestion
      name: rundeck.asc
```

Joris29 replied:
Alright i will add some tests for the new code and create a separate PR for repo thing

kenyon reviewed (commented):
The new code looks fine, but looks like we need to get the tests passing.

Joris29 commented:
@kenyon Added unit tests and improved some old test also changed the cli class itself a bit but still having issues with ubuntu repo gpg check

Joris29 commented:
@kenyon I Updated some things and had to remove CentOS 7 from unit tests as jq is not available by default.
Updated refs and readme i also created clean separate commits so those can added as is i think.
This only supports project config itself not project resources (yet). I will this i a follow up PR but i don't want to make it to big.

kenyon commented on manifests/init.pp:
```diff
@@ -106,6 +111,18 @@
 # @param service_script
 #   Allows you to use your own override template instead of the default from the package maintainer for rundeckd init script.
+# @param manage_cli
+#   Wheter to manage rundeck cli config and resource with the rundeck class or not.
```
typo fix

```suggestion
#   Whether to manage rundeck cli config and resource with the rundeck class or not.
```

kenyon reviewed (commented):
found one typo, otherwise looks good to me

Joris29 commented:
Fixed the typo
